### ISSUE TITLE
Failing to load on bpf-next: calipso_skbuff_delattr: opening perf event: cannot assign requested address

### ISSUE THREAD

**Issue by brb** [cilium/pwru#284]
```
Opening kprobe calipso_skbuff_delattr: creating perf_kprobe PMU (arch-specific fallback for "calipso_skbuff_delattr"): token calipso_skbuff_delattr: opening perf event: cannot assign requested address
```

From https://github.com/cilium/pwru/actions/runs/6719026297/job/18259892769?pr=283

**Comment by brb** [comment]
@lmb suggested that https://elixir.bootlin.com/linux/v6.6/source/kernel/trace/trace_kprobe.c#L884 this might be the case. Indeed, we are using symbols instead of addrs - https://github.com/cilium/pwru/blob/v1.0.4/main.go#L314.

**Comment by brb** [comment]
This is the new behavior introduced in 6.6 - https://lore.kernel.org/all/20231020104250.9537-1-flaniel@linux.microsoft.com/.

**Comment by lmb** [comment]
For now you could skip attaching if EADDRNOTAVAIL is generated and log a warning or something like that?

**Comment by brb** [comment]
>if EADDRNOTAVAIL is generated and log a warning or something like that

Yep, but we try to attach the same kprobe to multiple functions in a single go. We would need to figure out which function name is causing us troubles. I think we need to switch to addrs instead of syms.

**Comment by darox** [comment]
I patched my vm and now I'm facing the same issue:

```
./pwru 
2023/12/20 09:49:21 Attaching kprobes (via kprobe)...
712 / 2208 [------------------------>__________________________________________________] 32.25% ? p/s2023/12/20 09:49:21 Opening kprobe esp_input_restore_header: creating perf_kprobe PMU (arch-specific fallback for "esp_input_restore_header"): token esp_input_restore_header: opening perf event: cannot assign requested address
ubuntu# uname -a
Linux ubuntu 6.5.13-orbstack-00116-gb64875927ef8 #1 SMP Wed Dec  6 22:19:46 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```

**Comment by brb** [comment]
@darox We have released [v1.0.5](https://github.com/cilium/pwru/releases/tag/v1.0.5) which should have the tmp fix for the loading problem.

**Comment by darox** [comment]
I just rebased and it works. Thx!
